tests/NotificationTests/NotificationCenterTest.php

```diff
@@ -0,0 +1,639 @@
+        });
+
+        // add another callback for NotificationType::DECISION
+        $this->notificationCenterObj->addNotificationListener(NotificationType::DECISION, function () {
+        });
+
+        // Verify that notifications length for NotificationType::DECISION is 2
+        $this->assertSame(
+            2,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::DECISION])
+        );
+
+        // Verify that notifications length for NotificationType::TRACK is 1
+        $this->assertSame(
+            1,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::TRACK])
+        );
+
+        /////////////////////////////////////////////////////////////////////////////////////////
+        // === Verify that no notifications are removed given an invalid notification type === //
+        /////////////////////////////////////////////////////////////////////////////////////////
+
+        $invalid_type = "HelloWorld";
+
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(Logger::ERROR, "Invalid notification type.");
+
+        $this->assertSame(
+            null,
+            $this->notificationCenterObj->clearNotifications($invalid_type)
+        );
+
+        // Verify that notifications length for NotificationType::DECISION is still 2
+        $this->assertSame(
+            2,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::DECISION])
+        );
+
+        // Verify that notifications length for NotificationType::TRACK is still 1
+        $this->assertSame(
+            1,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::TRACK])
+        );
+
+        ///////////////////////////////////////////////////////////////////////////////////////
+        // === Verify that all notifications are removed given a valid notification type === //
+        ///////////////////////////////////////////////////////////////////////////////////////
+         
+        $this->loggerMock->expects($this->at(0))
+            ->method('log')
+            ->with(
+                Logger::INFO,
+                sprintf("All callbacks for notification type '%s' have been removed.", NotificationType::DECISION)
+            );
+
+        $this->notificationCenterObj->clearNotifications(NotificationType::DECISION);
+
+        // Verify that notifications length for NotificationType::DECISION is now 0
+        $this->assertSame(
+            0,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::DECISION])
+        );
+
+        // Verify that notifications length for NotificationType::TRACK is still 1
+        $this->assertSame(
+            1,
+            sizeof($this->notificationCenterObj->getNotifications()[NotificationType::TRACK])
+        );
+    }
+
+
+    public function testClearAllNotifications()
```

Comment:
on clean all, i would suggest the notificationid should be zero. what do u say?

Reply:
Can be. But making it zero doesn't provide additional benefit. In fact, if the customer stores these IDs as a reference in it's own memory, duplicate IDs may create an issue for them. So this way we ensure that IDs are always unqiue.

Reply:
make sense. this is what I was thinking.